avogadro/core/connectedgroup.cpp

```diff
@@ -0,0 +1,161 @@
+/******************************************************************************
+  This source file is part of the Avogadro project.
+  This source code is released under the 3-Clause BSD License, (see "LICENSE").
+******************************************************************************/
+
+#include "connectedgroup.h"
+
+#include <algorithm>
+#include <cassert>
```

Comment:
No asserts. If you want to add in something that prints warnings / errors to stdout or stderr, that's fine.

Reply:
Asserts can crash the UI, which is extremely confusing to an end-user.